src/connectors/search-box/connectSearchBox.ts

```diff
@@ -138,10 +138,10 @@ const connectSearchBox: SearchBoxConnector = function connectSearchBox(
         };
       },
 
-      getWidgetRenderState({ helper, searchMetadata }) {
+      getWidgetRenderState({ helper, searchMetadata, state }) {
         if (!_refine) {
           const setQueryAndSearch = (query: string) => {
-            if (query !== helper.state.query) {
+            if (query !== state.query) {
```

Comment:
this is what caused the bug (state is stale in this callback), but the part of code isn't actually needed, we can still set the query and search without condition. Fixing in #4990